torch/optim/optimizer.py

```diff
@@ -834,6 +842,14 @@ def register_load_state_dict_post_hook(
     @torch._disable_dynamo
     def load_state_dict(self, state_dict: StateDict) -> None:
         r"""Load the optimizer state.
+        The names of the parameters (if they exist in :meth:`state_dict`) will not affect the loading process.
```

Comment:
Put this entire explanation (whcih is very good btw!) under a Note so people know it's not the main carb of the function